actnow/accounts/permissions.py

```diff
@@ -1,8 +1,20 @@
 from oauth2_provider.models import get_application_model
-from rest_framework.permissions import BasePermission
+from rest_framework.permissions import SAFE_METHODS, BasePermission
 
 
 class AllowAppicationOwnerOnly(BasePermission):
     def has_permission(self, request, view):
         Application = get_application_model()
         return Application.objects.filter(user=request.user.id).exists()
+
+
+class IsOwnerOrReadOnly(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS:
+            return bool(request.user.is_active and request.user.is_authenticated)
+        return bool(obj.id == request.user.id)
+
+
+class IsNotDeleted(BasePermission):
+    def has_permission(self, request, view):
+        return bool(request.user.is_active and not request.user.is_deleted)
```

Comment:
If you're active, then it means you're already not deleted i.e. deleting a user should also mark `is_active` false.

Reply:
Have we updated this?

Reply:
yes.